src/librsync.h

```diff
@@ -37,9 +37,36 @@
 #define _RSYNC_H
 
 #include <sys/types.h>
-#include "types.h"
 #include "librsync-config.h"
 
+#define HAVE_UINT64 1 /* assume success, but might be undefined below */
+
+#if defined(HAVE_STDINT_H)
+#  include <stdint.h>
+#else
+# if SIZEOF_UNSIGNED_INT == 4
```

Comment:
Aren't all the defines that these depend on all defined in config.h? And isn't config.h in the same boat as types.h when it comes to what is deployed on systems?

I think librsync-config.h is deployed on systems and built by cmake specifically to avoid any dependencies on config.h.